### PR TITLE
Copy to clipboard with link instead of button

### PR DESCRIPTION
**Pull request by Pytal**
Addendum to https://github.com/nextcloud/server/pull/35240

Based on feedback from:
- https://github.com/nextcloud/server/issues/35505#issuecomment-1331743638
- https://github.com/nextcloud/server/pull/35240#issuecomment-1328003979

Requires:
- [ ] https://github.com/nextcloud/nextcloud-vue/pull/3532

### CODE REVIEW

File: apps/files_sharing/src/components/SharingEntryLink.vue [hunk 1]
```diff
@@ -38,14 +38,15 @@
 		<NcActions v-if="share && !isEmailShareType && share.token"
 			ref="copyButton"
 			class="sharing-entry__copy">
-			<NcActionButton :aria-label="t('files_sharing', 'Copy public link to clipboard')"
+			<NcActionLink :aria-label="t('files_sharing', 'Copy public link to clipboard')"
+				:href="shareLink"
```
Comment (skjnldsv) [inline apps/files_sharing/src/components/SharingEntryLink.vue:42]: We can also use `role="button"` if really necessary

**Review by skjnldsv — APPROVED** [review]

**Comment by skjnldsv** [comment]
@Pytal Generally speaking, please use the revert action in those cases. So it's clear that this is a revert of a previous commit. 
https://github.com/nextcloud/server/pull/35533

**Review by skjnldsv — CHANGES_REQUESTED** [review]
@Pytal Generally speaking, please use the revert action in those cases. So it's clear that this is a revert of a previous commit. 
https://github.com/nextcloud/server/pull/35533

File: apps/files_sharing/src/components/SharingEntryLink.vue (hunk 1 above)
Comment (Pytal) [reply apps/files_sharing/src/components/SharingEntryLink.vue:42]: Needs https://github.com/nextcloud/nextcloud-vue/pull/3532

**Comment by Pytal** [comment]
> @Pytal Generally speaking, please use the revert action in those cases. So it's clear that this is a revert of a previous commit. #35533

As described in the description this is an addendum to https://github.com/nextcloud/server/pull/35240 not a revert, as it keeps the material design icons

**Comment by skjnldsv** [comment]
> As described in the description this is an addendum to https://github.com/nextcloud/server/pull/35240 not a revert, as it keeps the material design icons

What about the tests ?

**Comment by Pytal** [comment]
> What about the tests ?

Since it's in one commit then reverting would also revert what we want to keep, would you suggest reverting the commit then re-adding the changes we want to keep?

**Comment by skjnldsv** [comment]
Generally speaking, I think it's more clear, when someone is out of context, what really happened if we revert a breaking PR.
Then you can open another PR and add back the changes for the material design icon.
Which are actually not at all what the original PR was designed to change.

If I have to pinpoint something later on and move up the blame history, not being clear that this PR was a revert because of a regression, it's going to make it harder.

Anyway... Not going to debate that further :grin:

**Comment by Pytal** [comment]
Closing in favour of https://github.com/nextcloud/server/pull/35533 + followup PR with new material design icons

**Pull request closed without merging**